pkgs/build-support/dotnet/make-nuget-deps/default.nix

```
@@ -1,23 +1,43 @@
-{ symlinkJoin
-, fetchurl
-, stdenvNoCC
-, lib
-, unzip
-, patchNupkgs
-, nugetPackageHook
-, fetchNupkg
+{
+  symlinkJoin,
+  lib,
+  fetchNupkg,
 }:
-lib.makeOverridable(
-  { name
-  , nugetDeps ? import sourceFile
-  , sourceFile ? null
-  , installable ? false
+lib.makeOverridable (
+  {
+    name,
+    nugetDeps ? null,
+    sourceFile ? null,
+    installable ? false,
   }:
+  let
+    callNupkg = pkg: if lib.isDerivation pkg then pkg else fetchNupkg pkg;
+
+    loadDeps =
+      x:
+      if x == null then
+        [ ]
```

```suggestion
        loadDeps sourceFile
```

Avoids the need for the if-then-else in `deps=` if we do it recursively here.

Could also go in the existing recursive section:

```nix
else loadDeps (
  if x == null then
    sourceFile
  else if isFunction x then
  #...
```

---

Honestly, I was considering if we couldn't just simplify this to accept just files instead.

Giving too many options is also an issue imo.

But I guess this could be simplified, I'll look into it.

---

> Giving too many options is also an issue imo.

True, although if some of those options have legitimate use-cases and they aren't too complex to implement & maintain then I'd rather be flexible.

For instance it may be feasible that users want to dynamically generate a list of deps themselves rather than using `nuget-to-*`... Although I don't know how realistic that example is?

I'm guessing this is similar to the non-list derivation case?

---

The non-list derivation case is for `mkNugetDeps` afaik.

If people want to use their own list of package derivations then they can use `mkNugetDeps`.